packages/button/src/Button.ts

```diff
@@ -127,11 +144,33 @@ export class Button extends SizedMixin(ButtonBase, { noDefaultSize: true }) {
     public set quiet(quiet: boolean) {
         this.treatment = quiet ? 'outline' : 'fill';
     }
-
     public get quiet(): boolean {
         return this.treatment === 'outline';
     }
 
+    protected override willUpdate(
+        changedProperties: PropertyValues<this>
+    ): void {
+        super.willUpdate(changedProperties);
+        if (changedProperties.has('pending')) {
+            if (this.pending) {
+                if (!this.disabled) {
+                    this.setAttribute('aria-label', this.pendingLabel);
+                }
+            } else if (
+                changedProperties.get('pending') !== undefined &&
+                !this.pending
```

Comment:
What do you see us addressing by the more complex condition here?

Reply:
If a consumer only provides the `aria-label` property for a button and doesn't set the label then this condition would make sure we won't override the `aria-label` value to `this.label` which is undefined in this case. 
Note: `sp-split-button` element currently does the same thing... we provide the `aria-label` to the buttons but not the `label`.

Reply:
Help me understand. How I currently understand the code as written:
```
if ("pending has changed") {
    if ("is pending") {
        if ("is not disabled") {
            // Apply pending label without regard for whether it is present or not
        }
    } else if ("is NOT pending AND NOT previously `undefined`, which happens at construction time when creating the `pending` property to begin with") {
        // Apply `label` or ''
    }
}
```
I like the idea of covering for when the author has applied `aria-label`, but I'm not sure that this code does. I _think_ doing so would look more like:
```
if ("pending has changed") {
    if ("is pending") {
        if ("is not disabled") {
            // Assume the applied `aria-label`, no matter who set it, is correct (this _might_ hit some issues if a Button was created as `pending` to start with, depending on code order)
            this.cachedLabel = this.getAttribute('aria-label');
            this.setAttribute('aria-label', this.pendingLabel);
        } else if (this.cachedLabel) {
            // If there was previously an applied `aria-label` re-apply it, or do nothing
            this.setAttribute('aria-label', this.cachedLabel);
        }
    }
}
```
This logic then duplicates some complexity with the change logic for `disabled`, which might benefit from being merged in, but is more annoying.

Either way, this got complex fast, it will be important to confirm the value of the `arial-label` attribute across these various states in the unit tests.
- pending===false => pending===true
- disabled===true && pending===false => disabled===true && pending===false
- disabled===false && pending===true => disabled===true && pending===true
- disabled===true && pending===true => disabled===false && pending===true
- All the same again with `aria-label` applied from the outside
- All the same again with `label` applied from the outside

There may be others, so think through the logic your creating and make sure the various workflows are fully managed so that the output you're expecting at the various points is specifically under test in the tests you add.

Reply:
I've updated the logic and added some new tests. I think they cover all of the possibilities (?) However the logic is now a bit complicated so if you need help in understanding anything or if I missed some edge-cases do let me know.

Reply:
@Westbrook  I've updated the code based on our discussion. Have a look whenever you get time